src/overlays/actors/ovl_En_Giant/z_en_giant.c

```diff
@@ -26,42 +31,449 @@ const ActorInit En_Giant_InitVars = {
+    }
+}
+
+s32 EnGiant_IsNotFreed(EnGiant* this) {
+    switch (GIANT_TYPE(&this->actor)) {
+        case GIANT_TYPE_SWAMP_TERMINA_FIELD:
+        case GIANT_TYPE_SWAMP_CLOCK_TOWER_SUCCESS:
+        case GIANT_TYPE_SWAMP_GIANTS_CHAMBER_AND_ENDING:
+        case GIANT_TYPE_SWAMP_CLOCK_TOWER_FAILURE:
+            if (!CHECK_QUEST_ITEM(0)) {
+                return 1;
+            }
+            break;
+        case GIANT_TYPE_MOUNTAIN_TERMINA_FIELD:
+        case GIANT_TYPE_MOUNTAIN_CLOCK_TOWER_SUCCESS:
+        case GIANT_TYPE_MOUNTAIN_GIANTS_CHAMBER_AND_ENDING:
+        case GIANT_TYPE_MOUNTAIN_CLOCK_TOWER_FAILURE:
+            if (!CHECK_QUEST_ITEM(1)) {
+                return 1;
+            }
+            break;
+        case GIANT_TYPE_OCEAN_TERMINA_FIELD:
+        case GIANT_TYPE_OCEAN_CLOCK_TOWER_SUCCESS:
+        case GIANT_TYPE_OCEAN_GIANTS_CHAMBER_AND_ENDING:
+        case GIANT_TYPE_OCEAN_CLOCK_TOWER_FAILURE:
+            if (!CHECK_QUEST_ITEM(2)) {
+                return 1;
+            }
+            break;
+        case GIANT_TYPE_CANYON_TERMINA_FIELD:
+        case GIANT_TYPE_CANYON_CLOCK_TOWER_SUCCESS:
+        case GIANT_TYPE_CANYON_GIANTS_CHAMBER_AND_ENDING:
+        case GIANT_TYPE_CANYON_CLOCK_TOWER_FAILURE:
+            if (!CHECK_QUEST_ITEM(3)) {
+                return 1;
+            }
+            break;
+    }
+
+    return 0;
+}
+
+void EnGiant_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnGiant* this = THIS;
+    s32 type = GIANT_TYPE(thisx);
+
+    this->actor.uncullZoneForward = 4000.0f;
+    this->actor.uncullZoneScale = 2000.0f;
+    this->actor.uncullZoneDownward = 2400.0f;
+    Actor_SetScale(&this->actor, 0.32f);
+    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_060079B0, &D_06002168, this->jointTable, this->morphTable, 16);
+    EnGiant_ChangeAnimation(this, GIANT_ANIMATION_IDLE_LOOP);
+    this->action = 0;
+    this->actionFunc = EnGiant_PerformCutsceneActions;
+    this->actor.draw = NULL;
+    this->alpha = 0;
+    this->actor.velocity.y = -10.0f;
+    this->actor.minVelocityY = -10.0f;
+    this->actor.gravity = -5.0f;
+    switch (type) {
+        case GIANT_TYPE_CANYON_TERMINA_FIELD:
+        case GIANT_TYPE_CANYON_CLOCK_TOWER_SUCCESS:
+        case GIANT_TYPE_CANYON_GIANTS_CHAMBER_AND_ENDING:
+            this->unk_24A = 0x1C6;
+            break;
+        case GIANT_TYPE_SWAMP_TERMINA_FIELD:
+        case GIANT_TYPE_SWAMP_CLOCK_TOWER_SUCCESS:
+        case GIANT_TYPE_SWAMP_GIANTS_CHAMBER_AND_ENDING:
+            this->unk_24A = 0x1C7;
+            break;
+        case GIANT_TYPE_OCEAN_TERMINA_FIELD:
+        case GIANT_TYPE_OCEAN_CLOCK_TOWER_SUCCESS:
+        case GIANT_TYPE_OCEAN_GIANTS_CHAMBER_AND_ENDING:
+            this->unk_24A = 0x1C8;
+            break;
+        default:
+            this->unk_24A = 0x1C5;
+            break;
+    }
+
+    if (type >= GIANT_TYPE_MOUNTAIN_CLOCK_TOWER_SUCCESS && type < GIANT_TYPE_MOUNTAIN_GIANTS_CHAMBER_AND_ENDING) {
```

Comment:
This looks like it could do with some defines. I'd suggest making a define for the whole condition, but I suspect that wouldn't be very popular.